vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -114,6 +114,41 @@ func TestAccResourceVSphereVirtualMachine(t *testing.T) {
 				},
 			},
 		},
+		{
+			"re-create on deletion",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereVirtualMachinePreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereVirtualMachineCheckExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereVirtualMachineConfigBasic(),
+						Check: resource.ComposeTestCheckFunc(
+							copyState(&state),
+							testAccResourceVSphereVirtualMachineCheckExists(true),
+						),
+					},
+					{
+						PreConfig: func() {
+							if err := testDeleteVM(state, "vm"); err != nil {
+								panic(err)
+							}
```

Comment:
maybe it's just and Azure provider thing - but we'd generally make the deletion a Check function and assert it's been deleted, rather than crashing?

Reply:
`PreConfig` does not return error, so the only possible thing we can do is panic here.

Reply:
Update: The intention here is to make sure this gets handled _completely_ outside of the plan/apply testing cycle. `PreConifg` happens [very early on](https://github.com/hashicorp/terraform/blob/master/helper/resource/testing.go#L736), which is why I use it - `Check` on the other hand happens pretty much in the middle [after the apply](https://github.com/hashicorp/terraform/blob/master/helper/resource/testing_config.go#L75), but before the post-apply refresh to check for an empty plan. Having these OOB steps in `PreConfig` kind of gives them a more real-world workflow (ie: someone usually makes manual modifications _between_ a TF run, versus in the middle of one).

What would be nicer is if the signature of `PreConfig` was altered to send an error back, but that's kind of a breaking change.